Tools/Pipeline/Gtk/Widgets/BuildOutput.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using Gtk;
+using Gdk;
+
+namespace MonoGame.Tools.Pipeline
+{
+    public partial class BuildOutput : Notebook
+    {
+        Pixbuf icon_begin_end = new Pixbuf (null, "MonoGame.Tools.Pipeline.Icons.build_begin_end.png");
+        Pixbuf icon_processing = new Pixbuf (null, "MonoGame.Tools.Pipeline.Icons.build_processing.png");
+        Pixbuf icon_fail = new Pixbuf (null, "MonoGame.Tools.Pipeline.Icons.build_fail.png");
+        Pixbuf icon_succeed = new Pixbuf (null, "MonoGame.Tools.Pipeline.Icons.build_succeed.png");
+        Pixbuf icon_skip = new Pixbuf (null, "MonoGame.Tools.Pipeline.Icons.build_skip.png");
+        Pixbuf icon_clean = new Pixbuf (null, "MonoGame.Tools.Pipeline.Icons.build_clean.png");
+
+        TreeIter tmpIter;
+        TreeStore listStore;
+
+        Uri folderUri;
+
+        public BuildOutput()
+        {
+            Build();
+
+            var column = new TreeViewColumn ();
+
+            var iconCell = new CellRendererPixbuf ();
+            var textCell = new CellRendererText ();
+            var idCell = new CellRendererText ();
+
+            idCell.Visible = false;
+
+            column.PackStart (iconCell, false);
+            column.PackStart (textCell, false);
+            column.PackStart (idCell, false);
+
+            treeview1.AppendColumn (column);
+
+            column.AddAttribute (iconCell,  "pixbuf", 0);
+            column.AddAttribute (textCell, "text", 1);
+            column.AddAttribute (idCell, "text", 2);
+
+            listStore = new TreeStore (typeof (Gdk.Pixbuf), typeof (string), typeof (string));
+
+            treeview1.Model = listStore;
+
+            textView1.SizeAllocated += TextView1_SizeAllocated;
+        }
+
+        internal void SetBaseFolder(IController controller)
+        {
+            string pl = ((PipelineController)controller).ProjectLocation;
+            if (!pl.EndsWith(System.IO.Path.DirectorySeparatorChar.ToString()))
+                pl += System.IO.Path.DirectorySeparatorChar;
+            folderUri = new Uri(pl);
+        }
+
+        private void TextView1_SizeAllocated (object o, SizeAllocatedArgs args)
+        {
+            textView1.ScrollToIter(textView1.Buffer.EndIter, 0, false, 0, 0);
+        }
+
+        public void OutputAppend(string text)
+        {
+            lock (textView1.Buffer)
+            {
+                textView1.Buffer.Text += text + "\r\n";
+            }
+
+            if (string.IsNullOrWhiteSpace(text))
+                return;
+
+            if (text.StartsWith("Build "))
+                AddItem(icon_begin_end, text.Remove(text.Length - 1), "");
```

Comment:
I see you use this to remove the period from the last build and then append the elapsed time.
It also removes one character from the first build message, instead of "10:23:34 PM" it shows  "10:23:34 P".
Instead I trimed the period only when I need to appent the time, https://github.com/mono/MonoGame/pull/4272/files#diff-70e4e9aa432ee996c57064ed582baed2R62